packages/odyssey-react-mui/src/OktaUiShell.tsx

```diff
@@ -0,0 +1,94 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { memo, useEffect, useState, type ReactNode } from "react";
+
+import { SideNav, type SideNavProps } from "./labs/SideNav";
+import { TopNav, type TopNavProps } from "./labs/TopNav";
+import { OdysseyProvider } from "./OdysseyProvider";
+import { ShadowDomElements } from "./shadow-dom";
+
+const containerStyles = {
+  display: "flex",
+};
+
+export type OktaUiShellComponentProps = {
+  sideNavProps: SideNavProps;
+  topNavProps: TopNavProps;
+};
+
+export type OktaUiShellProps = {
+  appComponent?: ReactNode;
+  onSubscriptionCreated: () => void;
+  subscribeToPropChanges: (
+    subscription: (componentProps: OktaUiShellComponentProps) => void,
+  ) => () => void;
+} & ShadowDomElements;
+
+export const defaultComponentProps: OktaUiShellComponentProps = {
+  sideNavProps: {
+    navHeaderText: "",
+    sideNavItems: [],
+  },
+  topNavProps: {
+    topNavLinkItems: [],
+  },
+};
+
+export const defaultProps = {
+  appComponent: <slot />,
+  onSubscriptionCreated: () => {},
+};
+
+const OktaUiShell = ({
+  appComponent = defaultProps.appComponent,
+  appRootElement,
+  emotionRootElement,
+  onSubscriptionCreated = defaultProps.onSubscriptionCreated,
+  subscribeToPropChanges,
+}: OktaUiShellProps) => {
+  const [componentProps, setComponentProps] = useState(defaultComponentProps);
+
+  useEffect(() => {
+    const unsubscribe = subscribeToPropChanges((componentProps) => {
+      setComponentProps(componentProps);
+    });
+
+    onSubscriptionCreated();
```

Comment:
This subscription was added to notify `renderReactInWebComponent` that the React app rendered.

I will move this to a generic wrapper component in `renderReactInWebComponent` so it's not required in the future when using components that aren't `OktaUiShell`.